## Calculating the Expected Return of a Stock (CAPM)

*Suggested Answers follow (usually there are multiple ways to solve a problem in Python).*

Obtain data for Microsoft and S&P 500 for the period 1st of January 2012 – 31st of December 2016 from Yahoo Finance.
Let S&P 500 act as the market.
Calculate the beta of Microsoft.


In [1]:


import numpy as np
import pandas as pd
import yfinance as yf
from pandas_datareader import data as wb

tickers = ['MSFT', '^GSPC']
data = pd.DataFrame()

for a in tickers:
    data[a] = yf.download(a, start='2012-1-1', end='2016-12-31', auto_adjust=False)['Adj Close']


sec_returns = np.log( data / data.shift(1) )
cov = sec_returns.cov() * 250
cov_with_market = cov.iloc[0,1]
market_var = sec_returns['^GSPC'].var() * 250

MSFT_beta = cov_with_market / market_var

[*********************100%***********************]  1 of 1 completed
[*********************100%***********************]  1 of 1 completed


Assume a risk-free rate of 2.5% and a risk premium of 5%. <br />
Estimate the expected return of Microsoft.

**Calculate the expected return of P&G (CAPM):**
### $$
\overline{r_{pg}} = r_f + \beta_{pg}(\overline{r_{m}} - r_f)
$$

In [2]:
MSFT_er = 0.025 + MSFT_beta * 0.05
MSFT_er

np.float64(0.08065701544402351)

Kết quả 8.07% là tỷ suất sinh lợi tối thiểu mà một nhà đầu tư nên kỳ vọng nhận được từ cổ phiếu Microsoft (trong giai đoạn 2012-2016 với các giả định trên) để bù đắp cho rủi ro mà họ gánh chịu.

Họ nhận 2.5% vì đã bỏ tiền ra đầu tư (thay vì giữ tiền mặt).

Họ nhận thêm 5.57% (tức là 1.113∗5%) làm phần bù rủi ro.

Tại sao phần bù của MSFT (5.57%) lại cao hơn phần bù của thị trường (5%)?

Bởi vì Beta của MSFT là 1.113 (lớn hơn 1).

Điều này có nghĩa là MSFT rủi ro hơn, biến động mạnh hơn thị trường. Vì nó rủi ro hơn, nhà đầu tư phải được "thưởng" nhiều hơn để nắm giữ nó. Nếu Beta nhỏ hơn 1, phần bù rủi ro của cổ phiếu sẽ thấp hơn 5%.